framework/src/base/MooseApp.C

```diff
@@ -516,6 +516,20 @@ MooseApp::setupOptions()
     }
 
     _parser.parse(_input_filename);
+
+    if (isParamValid("mesh_only"))
+    {
+      // Clear out the normal MOOSE tasks. We'll repopulate the syntax object with a subset
+      // of tasks necessary for processing the mesh.
+      _syntax.clearTaskDependencies();
+
+      // Call the global function in Moose.C to prepopulate mesh-only tasks
+      Moose::populateMeshOnlyTasks(_syntax);
+
+      // Call the user-definable callback
+      modifyMeshOnlyTasks(_syntax);
```

Comment:
Here's the new callback where you are passed a reference to the `Syntax` object. You can then make the normal calls to modify the task dependencies as needed. Will this work for your needs in Rattlesnake?

Reply:
Yes, it will. The only thing is that we need to maintain two separate lists of tasks. My way assumes the mesh-only tasks are always in the same order of the tasks in the normal calculations. This may not necessarily a bad thing. I think this is fine to me.

Reply:
I mean your way may not be a bad thing.